src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/ConstantExpectedAnalyzer.cs

```diff
@@ -0,0 +1,566 @@
+                    {
+                        context.ReportDiagnostic(parameterCheckDiagnostic);
+                    }
+                    continue;
+                }
+                var constantValue = v.ConstantValue;
+                if (!constantValue.HasValue)
+                {
+                    var location = argument.Syntax.GetLocation();
+                    context.ReportDiagnostic(Diagnostic.Create(CA1861.ConstantNotConstantRule, location));
+                    continue;
+                }
+
+                var rawValue = constantValue.Value;
+                if (!argConstantParameter.ValidateValue(argument, rawValue, out var valueDiagnostic))
+                {
+                    context.ReportDiagnostic(valueDiagnostic);
+                }
+            }
+        }
+
+        protected abstract void RegisterAttributeSyntax(CompilationStartAnalysisContext context);
+
+        protected void OnParameterWithConstantExpectedAttribute(IParameterSymbol parameter, Action<Diagnostic> reportAction)
+        {
+            if (!ValidateConstantExpectedParameter(parameter, out ImmutableArray<Diagnostic> diagnostics))
+            {
+                foreach (var diagnostic in diagnostics)
+                {
+                    reportAction(diagnostic);
+                }
+            }
+        }
+
+        private static bool TryCreateConstantExpectedParameter(IParameterSymbol parameterSymbol, [NotNullWhen(true)] out ConstantExpectedParameter? parameter)
+        {
+            if (!TryGetConstantExpectedAttributeData(parameterSymbol, out var attributeData))
+            {
+                parameter = null;
+                return false;
+            }
+
+            switch (parameterSymbol.Type.SpecialType)
+            {
+                case SpecialType.System_Char:
+                    return UnmanagedHelper<char>.TryCreate(parameterSymbol, attributeData, char.MinValue, char.MaxValue, out parameter);
+                case SpecialType.System_Byte:
+                    return UnmanagedHelper<ulong>.TryCreate(parameterSymbol, attributeData, byte.MinValue, byte.MaxValue, out parameter);
+                case SpecialType.System_UInt16:
+                    return UnmanagedHelper<ulong>.TryCreate(parameterSymbol, attributeData, ushort.MinValue, ushort.MaxValue, out parameter);
+                case SpecialType.System_UInt32:
+                    return UnmanagedHelper<ulong>.TryCreate(parameterSymbol, attributeData, uint.MinValue, uint.MaxValue, out parameter);
+                case SpecialType.System_UInt64:
+                    return UnmanagedHelper<ulong>.TryCreate(parameterSymbol, attributeData, ulong.MinValue, ulong.MaxValue, out parameter);
+                case SpecialType.System_UIntPtr:
+                    return UnmanagedHelper<ulong>.TryCreate(parameterSymbol, attributeData, uint.MinValue, uint.MaxValue, out parameter);
```

Comment:
@tannergooding In your opinion, is it a good practice to check for uint.MinValue and uint.MaxValue, since UIntPtr has a platform-dependent size?